Permanent/ViewControllers/LocationSetViewController.swift

```diff
@@ -0,0 +1,222 @@
+            } else {
+                self.view.showNotificationBanner(title: "There was a problem saving the location.".localized(), backgroundColor: .deepRed, textColor: .white)
+            }
+        })
+    }
+    
+    func getLocationString(_ items: [String?]) -> String {
+        return items.compactMap { $0 }.joined(separator: ", ")
+    }
+}
+extension LocationSetViewController: MKMapViewDelegate {
+    func mapView(_ mapView: MKMapView, didAdd views: [MKAnnotationView]) {
+        for item in views {
+            item.isEnabled = false
+        }
+    }
+}
+extension LocationSetViewController: AutoCompletionTextFieldDataSource {
+    func fetchSuggestions(forIncompleteString incompleteString: String!, withCompletionBlock completion: FetchCompletionBlock!) {
+        if incompleteString.count > 2 {
+            geocoder.geocodeAddressString(incompleteString,
+                                          completionHandler: { (result, error) in
+                                            if let placemarks = result {
+                                                let placemarksDictionary = placemarks.map { placemark -> [String:String] in
+                                                    let addressElements: [String?] = [placemark.thoroughfare, placemark.subThoroughfare, placemark.locality, placemark.country]
+                                                    let addressString = self.getLocationString(addressElements)
+                                                    self.locationDetails = addressString
+                                                    if let coordinate = placemark.location?.coordinate,
+                                                       let radius = placemark.region as? CLCircularRegion {
+                                                        self.searchedLocations[addressString] = (coordinate,radius.radius)
+                                                    }
+                                                    return ["title": addressString]
+                                                }
+                                                completion(placemarksDictionary, "title")
+                                            }
+                                          })
+        }
+    }
+}
+
+extension LocationSetViewController: AutoCompletionTextFieldDelegate {
+    func textField(_ textField: AutoCompletionTextField!, didSelectItem selectedItem: Any!) {
+        if let item = selectedItem as? [String: String] {
+            let itemTitle: String = item["title"] ?? ""
+            searchBar.text = itemTitle
+            if let location: CLLocationCoordinate2D = self.searchedLocations[itemTitle]?.0,
+               let radius: Double = self.searchedLocations[itemTitle]?.1 {
+                setMapRegion(location,radius)
+                saveLocation(location)
+            }
+            self.view.endEditing(true)
+        }
+    }
+}
+
+extension LocationSetViewController: AutoCompletionAnimator {
```

Comment:
this animator delegate is not used. safe to delete

Reply:
the delegate will be removed.